### ISSUE TITLE
Question: Using Vtorc，Primary is Dead，how can i do to let it failover normally

### ISSUE THREAD

**Issue by shizhibo2016** [vitessio/vitess#12657]
### Question

I kill -9 the mysql master, but it can not failover normally.

ERROR LOG:
                     E0318 06:32:01.683227 51275 topology_recovery.go:800] AttemptRecoveryRegistration: keyspace keyspace_yy shard 0 has recently experienced a failover (of 10.1.9.91:17201) and is in active period. It will not be failed over again. You may acknowledge the failure on this cluster (-c ack-cluster-recoveries) or on 10.1.9.91:17201 (-c ack-instance-recoveries) to remove this blockage

**Comment by isredstar** [comment]
Yes，I also occur the same question

**Comment by isredstar** [comment]
**repeat the problem**
```
[vitess@m115 ~]$ ps -ef|grep mysqld
vitess   1629163       1  0 10:54 pts/34   00:00:10 /usr/sbin/mysqld --defaults-file=/tmp/vtdataroot/vt_0000000100/my.cnf --basedir=/usr
vitess   1629339       1  0 10:54 pts/34   00:00:10 /usr/sbin/mysqld --defaults-file=/tmp/vtdataroot/vt_0000000101/my.cnf --basedir=/usr
vitess   1629551       1  0 10:54 pts/34   00:00:10 /usr/sbin/mysqld --defaults-file=/tmp/vtdataroot/vt_0000000102/my.cnf --basedir=/usr
vitess   1643053       1  0 11:06 pts/34   00:00:07 /usr/sbin/mysqld --defaults-file=/tmp/vtdataroot/vt_0000000300/my.cnf --basedir=/usr
vitess   1643215       1  0 11:06 pts/34   00:00:06 /usr/sbin/mysqld --defaults-file=/tmp/vtdataroot/vt_0000000301/my.cnf --basedir=/usr
vitess   1643426       1  0 11:06 pts/34   00:00:06 /usr/sbin/mysqld --defaults-file=/tmp/vtdataroot/vt_0000000302/my.cnf --basedir=/usr
vitess   1643662       1  0 11:06 pts/34   00:00:07 /usr/sbin/mysqld --defaults-file=/tmp/vtdataroot/vt_0000000400/my.cnf --basedir=/usr
vitess   1643881       1  0 11:06 pts/34   00:00:06 /usr/sbin/mysqld --defaults-file=/tmp/vtdataroot/vt_0000000401/my.cnf --basedir=/usr
vitess   1644055       1  0 11:07 pts/34   00:00:06 /usr/sbin/mysqld --defaults-file=/tmp/vtdataroot/vt_0000000402/my.cnf --basedir=/usr
```

<html>
<body>


Keyspace | Shard | Alias | Type | Tablet State | Hostname | Actions
-- | -- | -- | -- | -- | -- | --
commercelocal | 0 | zone1-100 | PRIMARY | SERVING | localhost |  
commercelocal | 0 | zone1-102 | RDONLY | SERVING | localhost |  
commercelocal | 0 | zone1-101 | REPLICA | SERVING | localhost |  
customerlocal | -80 | zone1-300 | PRIMARY | NOT_SERVING | localhost |  
customerlocal | -80 | zone1-302 | RDONLY | SERVING | localhost |  
customerlocal | -80 | zone1-301 | REPLICA | SERVING | localhost


</body>
</html>

**# kill node 300**
`kill -9 1643053`

**vtorc.out**
```
I0321 11:34:45.608782 1629626 topology_recovery.go:610] executeCheckAndRecoverFunction: proceeding with DeadPrimary detection on localhost:17300; isActionable?: true; skipProcesses: false
I0321 11:34:45.611078 1629626 topology_recovery.go:427] topology_recovery: detected DeadPrimary failure on localhost:17300
I0321 11:34:45.613676 1629626 locks.go:432] Locking shard customer/-80 for action Orc Recovery
I0321 11:34:45.627852 1629626 tablet_discovery.go:169] force refresh of all tablets in shard - customer/-80
I0321 11:34:45.635051 1629626 tablet_discovery.go:234] Discovered: alias:{cell:"zone1" uid:300} hostname:"localhost" port_map:{key:"grpc" value:16300} port_map:{key:"vt" value:15300} keyspace:"customer"
 shard:"-80" key_range:{end:"\x80"} type:PRIMARY mysql_hostname:"localhost" mysql_port:17300 primary_term_start_time:{seconds:1679368011 nanoseconds:777687607} db_server_version:"5.7.41" default_conn_collation:255I0321 11:34:45.635849 1629626 tablet_discovery.go:234] Discovered: alias:{cell:"zone1" uid:302} hostname:"localhost" port_map:{key:"grpc" value:16302} port_map:{key:"vt" value:15302} keyspace:"customer"
 shard:"-80" key_range:{end:"\x80"} type:RDONLY mysql_hostname:"localhost" mysql_port:17302 db_server_version:"5.7.41" default_conn_collation:255I0321 11:34:45.636593 1629626 tablet_discovery.go:234] Discovered: alias:{cell:"zone1" uid:301} hostname:"localhost" port_map:{key:"grpc" value:16301} port_map:{key:"vt" value:15301} keyspace:"customer"
 shard:"-80" key_range:{end:"\x80"} type:REPLICA mysql_hostname:"localhost" mysql_port:17301 db_server_version:"5.7.41" default_conn_collation:255I0321 11:34:45.654762 1629626 orchestrator.go:238] Force discovered - <nil>, err - rpc error: code = Unknown desc = TabletManager.FullStatus on zone1-0000000300 error: net.Dial(/tmp/vtdataroot/vt_000000
0300/mysql.sock) to local server failed: dial unix /tmp/vtdataroot/vt_0000000300/mysql.sock: connect: connection refused (errno 2002) (sqlstate HY000): net.Dial(/tmp/vtdataroot/vt_0000000300/mysql.sock) to local server failed: dial unix /tmp/vtdataroot/vt_0000000300/mysql.sock: connect: connection refused (errno 2002) (sqlstate HY000)W0321 11:34:45.654820 1629626 orchestrator.go:252]  DiscoverInstance(localhost:17300) instance is nil in 0.017s (Backend: 0.006s, Instance: 0.011s), error=rpc error: code = Unknown desc = TabletManager.
FullStatus on zone1-0000000300 error: net.Dial(/tmp/vtdataroot/vt_0000000300/mysql.sock) to local server failed: dial unix /tmp/vtdataroot/vt_0000000300/mysql.sock: connect: connection refused (errno 2002) (sqlstate HY000): net.Dial(/tmp/vtdataroot/vt_0000000300/mysql.sock) to local server failed: dial unix /tmp/vtdataroot/vt_0000000300/mysql.sock: connect: connection refused (errno 2002) (sqlstate HY000)I0321 11:34:45.689590 1629626 orchestrator.go:238] Force discovered - &{Key:localhost:17301 InstanceAlias:zone1-0000000301 ServerID:1776132157 ServerUUID:6750ddf8-c795-11ed-9cce-000c296af92f Version:5.7
.41 VersionComment:MySQL Community Server (GPL) FlavorName: ReadOnly:true BinlogFormat:ROW BinlogRowImage:FULL LogBinEnabled:true LogReplicationUpdatesEnabled:true SelfBinlogCoordinates:vt-0000000301-bin.000001:75543 SourceKey:localhost:17300 SourceUUID:64d59093-c795-11ed-93d1-000c296af92f AncestryUUID:64d59093-c795-11ed-93d1-000c296af92f,6750ddf8-c795-11ed-9cce-000c296af92f IsDetachedPrimary:false ReplicationSQLThreadRuning:true ReplicationIOThreadRuning:false ReplicationSQLThreadState:1 ReplicationIOThreadState:2 HasReplicationFilters:false GTIDMode:ON SupportsOracleGTID:true UsingOracleGTID:true UsingMariaDBGTID:false UsingPseudoGTID:false ReadBinlogCoordinates:vt-0000000300-bin.000001:76812 ExecBinlogCoordinates:vt-0000000300-bin.000001:76812 IsDetached:false RelaylogCoordinates:vt-0000000301-relay-bin.000002:77041 LastSQLError: LastIOError:"error reconnecting to master 'vt_repl@localhost:17300' - retry-time: 10  retries: 37" SecondsBehindPrimary:{Int64:0 Valid:false} SQLDelay:0 ExecutedGtidSet:64d59093-c795-11ed-93d1-000c296af92f:1-115 GtidPurged: GtidErrant: primaryExecutedGtidSet:64d59093-c795-11ed-93d1-000c296af92f:1-115 ReplicationLagSeconds:{Int64:0 Valid:false} DataCenter:zone1 Region: PhysicalEnvironment: ReplicationDepth:1 IsCoPrimary:false HasReplicationCredentials:true SemiSyncEnforced:false SemiSyncPrimaryEnabled:false SemiSyncReplicaEnabled:true SemiSyncPrimaryTimeout:1000000000000000000 SemiSyncPrimaryWaitForReplicaCount:1 SemiSyncPrimaryStatus:false SemiSyncPrimaryClients:0 SemiSyncReplicaStatus:true LastSeenTimestamp: IsLastCheckValid:true IsUpToDate:true IsRecentlyChecked:true SecondsSinceLastSeen:{Int64:0 Valid:false} IsCandidate:false PromotionRule:neutral IsDowntimed:false DowntimeReason: DowntimeOwner: DowntimeEndTimestamp: ElapsedDowntime:0s UnresolvedHostname: AllowTLS:false Problems:[] LastDiscoveryLatency:41.687422ms seed:false ReplicationGroupName: ReplicationGroupIsSinglePrimary:false ReplicationGroupMemberState: ReplicationGroupMemberRole: ReplicationGroupMembers:map[] ReplicationGroupPrimaryInstanceKey::0}, err - <nil>I0321 11:34:45.701248 1629626 orchestrator.go:238] Force discovered - &{Key:localhost:17302 InstanceAlias:zone1-0000000302 ServerID:1392133821 ServerUUID:69cb1981-c795-11ed-a7cc-000c296af92f Version:5.7
.41 VersionComment:MySQL Community Server (GPL) FlavorName: ReadOnly:true BinlogFormat:ROW BinlogRowImage:FULL LogBinEnabled:true LogReplicationUpdatesEnabled:true SelfBinlogCoordinates:vt-0000000302-bin.000001:75543 SourceKey:localhost:17300 SourceUUID:64d59093-c795-11ed-93d1-000c296af92f AncestryUUID:64d59093-c795-11ed-93d1-000c296af92f,69cb1981-c795-11ed-a7cc-000c296af92f IsDetachedPrimary:false ReplicationSQLThreadRuning:true ReplicationIOThreadRuning:false ReplicationSQLThreadState:1 ReplicationIOThreadState:2 HasReplicationFilters:false GTIDMode:ON SupportsOracleGTID:true UsingOracleGTID:true UsingMariaDBGTID:false UsingPseudoGTID:false ReadBinlogCoordinates:vt-0000000300-bin.000001:76812 ExecBinlogCoordinates:vt-0000000300-bin.000001:76812 IsDetached:false RelaylogCoordinates:vt-0000000302-relay-bin.000002:77041 LastSQLError: LastIOError:"error reconnecting to master 'vt_repl@localhost:17300' - retry-time: 10  retries: 37" SecondsBehindPrimary:{Int64:0 Valid:false} SQLDelay:0 ExecutedGtidSet:64d59093-c795-11ed-93d1-000c296af92f:1-115 GtidPurged: GtidErrant: primaryExecutedGtidSet:64d59093-c795-11ed-93d1-000c296af92f:1-115 ReplicationLagSeconds:{Int64:0 Valid:false} DataCenter:zone1 Region: PhysicalEnvironment: ReplicationDepth:1 IsCoPrimary:false HasReplicationCredentials:true SemiSyncEnforced:false SemiSyncPrimaryEnabled:false SemiSyncReplicaEnabled:false SemiSyncPrimaryTimeout:1000000000000000000 SemiSyncPrimaryWaitForReplicaCount:1 SemiSyncPrimaryStatus:false SemiSyncPrimaryClients:0 SemiSyncReplicaStatus:false LastSeenTimestamp: IsLastCheckValid:true IsUpToDate:true IsRecentlyChecked:true SecondsSinceLastSeen:{Int64:0 Valid:false} IsCandidate:false PromotionRule:must_not IsDowntimed:false DowntimeReason: DowntimeOwner: DowntimeEndTimestamp: ElapsedDowntime:0s UnresolvedHostname: AllowTLS:false Problems:[] LastDiscoveryLatency:55.256033ms seed:false ReplicationGroupName: ReplicationGroupIsSinglePrimary:false ReplicationGroupMemberState: ReplicationGroupMemberRole: ReplicationGroupMembers:map[] ReplicationGroupPrimaryInstanceKey::0}, err - <nil>I0321 11:34:45.709150 1629626 topology_recovery.go:706] executeCheckAndRecoverFunction: proceeding with DeadPrimary recovery on localhost:17300; isRecoverable?: true; skipProcesses: false
I0321 11:34:45.713106 1629626 topology_recovery.go:275] Analysis: DeadPrimary, deadprimary localhost:17300
I0321 11:34:45.713990 1629626 topology_recovery.go:293] ERS - will initiate emergency reparent shard in keyspace - customer, shard - -80
I0321 11:34:45.714027 1629626 topology_recovery.go:209] topology_recovery: will initiate emergency reparent shard in keyspace - customer, shard - -80
I0321 11:34:45.715248 1629626 topology_recovery.go:293] ERS - Getting a new durability policy for semi_sync
I0321 11:34:45.715294 1629626 topology_recovery.go:209] topology_recovery: Getting a new durability policy for semi_sync
I0321 11:34:45.719038 1629626 topology_recovery.go:293] ERS - getting replication position from zone1-0000000301
I0321 11:34:45.719084 1629626 topology_recovery.go:209] topology_recovery: getting replication position from zone1-0000000301
I0321 11:34:45.720040 1629626 topology_recovery.go:293] ERS - getting replication position from zone1-0000000302
I0321 11:34:45.720154 1629626 topology_recovery.go:209] topology_recovery: getting replication position from zone1-0000000302
I0321 11:34:45.720973 1629626 topology_recovery.go:293] ERS - getting replication position from zone1-0000000300
I0321 11:34:45.721013 1629626 topology_recovery.go:209] topology_recovery: getting replication position from zone1-0000000300
W0321 11:34:45.731469 1629626 topology_recovery.go:289] ERS - failed to get replication status from zone1-0000000300: rpc error: code = Unknown desc = TabletManager.StopReplicationAndGetStatus on zone1-
0000000300 error: before status failed: net.Dial(/tmp/vtdataroot/vt_0000000300/mysql.sock) to local server failed: dial unix /tmp/vtdataroot/vt_0000000300/mysql.sock: connect: connection refused (errno 2002) (sqlstate HY000): before status failed: net.Dial(/tmp/vtdataroot/vt_0000000300/mysql.sock) to local server failed: dial unix /tmp/vtdataroot/vt_0000000300/mysql.sock: connect: connection refused (errno 2002) (sqlstate HY000)I0321 11:34:45.731518 1629626 topology_recovery.go:209] topology_recovery: failed to get replication status from zone1-0000000300: rpc error: code = Unknown desc = TabletManager.StopReplicationAndGetSta
tus on zone1-0000000300 error: before status failed: net.Dial(/tmp/vtdataroot/vt_0000000300/mysql.sock) to local server failed: dial unix /tmp/vtdataroot/vt_0000000300/mysql.sock: connect: connection refused (errno 2002) (sqlstate HY000): before status failed: net.Dial(/tmp/vtdataroot/vt_0000000300/mysql.sock) to local server failed: dial unix /tmp/vtdataroot/vt_0000000300/mysql.sock: connect: connection refused (errno 2002) (sqlstate HY000)I0321 11:34:45.734012 1629626 topology_recovery.go:293] ERS - started finding the intermediate source
I0321 11:34:45.734058 1629626 topology_recovery.go:209] topology_recovery: started finding the intermediate source
I0321 11:34:45.734534 1629626 topology_recovery.go:293] ERS - finding intermediate source - sorted replica: cell:"zone1" uid:301
I0321 11:34:45.734566 1629626 topology_recovery.go:209] topology_recovery: finding intermediate source - sorted replica: cell:"zone1" uid:301
I0321 11:34:45.734908 1629626 topology_recovery.go:293] ERS - finding intermediate source - sorted replica: cell:"zone1" uid:302
I0321 11:34:45.734927 1629626 topology_recovery.go:209] topology_recovery: finding intermediate source - sorted replica: cell:"zone1" uid:302
I0321 11:34:45.735092 1629626 topology_recovery.go:293] ERS - intermediate source selected - cell:"zone1" uid:301
I0321 11:34:45.735115 1629626 topology_recovery.go:209] topology_recovery: intermediate source selected - cell:"zone1" uid:301
I0321 11:34:45.735260 1629626 topology_recovery.go:293] ERS - Removing zone1-0000000301 from list of valid candidates for promotion because it will not be able to make forward progress on promotion with
 the tablets currently reachableI0321 11:34:45.735266 1629626 topology_recovery.go:209] topology_recovery: Removing zone1-0000000301 from list of valid candidates for promotion because it will not be able to make forward progress on p
romotion with the tablets currently reachableI0321 11:34:45.735405 1629626 topology_recovery.go:293] ERS - Removing zone1-0000000302 from list of valid candidates for promotion because it has the Must Not promote rule
I0321 11:34:45.735420 1629626 topology_recovery.go:209] topology_recovery: Removing zone1-0000000302 from list of valid candidates for promotion because it has the Must Not promote rule
E0321 11:34:45.735636 1629626 topology_recovery.go:307] Error running ERS - Code: FAILED_PRECONDITION
no valid candidates for emergency reparent
I0321 11:34:45.735980 1629626 topology_recovery.go:723] Topology recovery: {"ID":365,"UID":"1679369685711765701:969824cf71f3e491092f1afc82d555b20cec202967f14d1f9a40c07fce8fade8","AnalysisEntry":{"Analyz
edInstanceKey":{"Hostname":"localhost","Port":17300},"AnalyzedInstancePrimaryKey":{"Hostname":"","Port":0},"TabletType":1,"PrimaryTimeStamp":"0001-01-01T00:00:00Z","ClusterDetails":{"Keyspace":"customer","Shard":"-80","CountInstances":0,"HeuristicLag":0,"HasAutomatedPrimaryRecovery":true,"HasAutomatedIntermediatePrimaryRecovery":true},"AnalyzedInstanceDataCenter":"zone1","AnalyzedInstanceRegion":"","AnalyzedKeyspace":"customer","AnalyzedShard":"-80","AnalyzedInstancePhysicalEnvironment":"","AnalyzedInstanceBinlogCoordinates":{"LogFile":"vt-0000000300-bin.000001","LogPos":76812,"Type":0},"IsPrimary":true,"IsClusterPrimary":true,"IsCoPrimary":false,"LastCheckValid":false,"LastCheckPartialSuccess":false,"CountReplicas":2,"CountValidReplicas":2,"CountValidReplicatingReplicas":0,"CountReplicasFailingToConnectToPrimary":2,"CountDowntimedReplicas":0,"ReplicationDepth":0,"IsFailingToConnectToPrimary":false,"ReplicationStopped":true,"Analysis":"DeadPrimary","Description":"Primary cannot be reached by vtorc and none of its replicas is replicating","StructureAnalysis":null,"IsDowntimed":false,"IsReplicasDowntimed":false,"DowntimeEndTimestamp":"","DowntimeRemainingSeconds":0,"IsBinlogServer":false,"OracleGTIDImmediateTopology":true,"MariaDBGTIDImmediateTopology":false,"BinlogServerImmediateTopology":false,"SemiSyncPrimaryEnabled":true,"SemiSyncPrimaryStatus":true,"SemiSyncPrimaryWaitForReplicaCount":1,"SemiSyncPrimaryClients":1,"SemiSyncReplicaEnabled":true,"CountSemiSyncReplicasEnabled":1,"CountLoggingReplicas":2,"CountStatementBasedLoggingReplicas":0,"CountMixedBasedLoggingReplicas":0,"CountRowBasedLoggingReplicas":2,"CountDistinctMajorVersionsLoggingReplicas":1,"CountDelayedReplicas":0,"CountLaggingReplicas":0,"IsActionableRecovery":true,"ProcessingNodeHostname":"m115","ProcessingNodeToken":"1265cc04983f427e4f1d435acae5585009ab6234efffc9e7a1205fcf33f987c3","CountAdditionalAgreeingNodes":0,"StartActivePeriod":"","SkippableDueToDowntime":false,"GTIDMode":"ON","MinReplicaGTIDMode":"ON","MaxReplicaGTIDMode":"ON","MaxReplicaGTIDErrant":"","CommandHint":"","IsReadOnly":false},"SuccessorKey":null,"SuccessorAlias":"","IsActive":false,"IsSuccessful":false,"LostReplicas":[],"ParticipatingInstanceKeys":[],"AllErrors":[],"RecoveryStartTimestamp":"","RecoveryEndTimestamp":"","ProcessingNodeHostname":"","ProcessingNodeToken":"","Acknowledged":false,"AcknowledgedAt":"","AcknowledgedBy":"","AcknowledgedComment":"","LastDetectionID":0,"RelatedRecoveryID":0,"Type":"","RecoveryType":"NotPrimaryRecovery"}I0321 11:34:45.736009 1629626 tablet_discovery.go:169] force refresh of all tablets in shard - customer/-80
I0321 11:34:45.740341 1629626 tablet_discovery.go:234] Discovered: alias:{cell:"zone1" uid:301} hostname:"localhost" port_map:{key:"grpc" value:16301} port_map:{key:"vt" value:15301} keyspace:"customer"
 shard:"-80" key_range:{end:"\x80"} type:REPLICA mysql_hostname:"localhost" mysql_port:17301 db_server_version:"5.7.41" default_conn_collation:255I0321 11:34:45.741336 1629626 tablet_discovery.go:234] Discovered: alias:{cell:"zone1" uid:302} hostname:"localhost" port_map:{key:"grpc" value:16302} port_map:{key:"vt" value:15302} keyspace:"customer"
 shard:"-80" key_range:{end:"\x80"} type:RDONLY mysql_hostname:"localhost" mysql_port:17302 db_server_version:"5.7.41" default_conn_collation:255I0321 11:34:45.745295 1629626 tablet_discovery.go:234] Discovered: alias:{cell:"zone1" uid:300} hostname:"localhost" port_map:{key:"grpc" value:16300} port_map:{key:"vt" value:15300} keyspace:"customer"
 shard:"-80" key_range:{end:"\x80"} type:PRIMARY mysql_hostname:"localhost" mysql_port:17300 primary_term_start_time:{seconds:1679368011 nanoseconds:777687607} db_server_version:"5.7.41" default_conn_collation:255I0321 11:34:45.749878 1629626 orchestrator.go:238] Force discovered - <nil>, err - rpc error: code = Unknown desc = TabletManager.FullStatus on zone1-0000000300 error: net.Dial(/tmp/vtdataroot/vt_000000
0300/mysql.sock) to local server failed: dial unix /tmp/vtdataroot/vt_0000000300/mysql.sock: connect: connection refused (errno 2002) (sqlstate HY000): net.Dial(/tmp/vtdataroot/vt_0000000300/mysql.sock) to local server failed: dial unix /tmp/vtdataroot/vt_0000000300/mysql.sock: connect: connection refused (errno 2002) (sqlstate HY000)I0321 11:34:45.769894 1629626 orchestrator.go:238] Force discovered - &{Key:localhost:17302 InstanceAlias:zone1-0000000302 ServerID:1392133821 ServerUUID:69cb1981-c795-11ed-a7cc-000c296af92f Version:5.7
.41 VersionComment:MySQL Community Server (GPL) FlavorName: ReadOnly:true BinlogFormat:ROW BinlogRowImage:FULL LogBinEnabled:true LogReplicationUpdatesEnabled:true SelfBinlogCoordinates:vt-0000000302-bin.000001:75543 SourceKey:localhost:17300 SourceUUID:64d59093-c795-11ed-93d1-000c296af92f AncestryUUID:64d59093-c795-11ed-93d1-000c296af92f,69cb1981-c795-11ed-a7cc-000c296af92f IsDetachedPrimary:false ReplicationSQLThreadRuning:true ReplicationIOThreadRuning:false ReplicationSQLThreadState:1 ReplicationIOThreadState:2 HasReplicationFilters:false GTIDMode:ON SupportsOracleGTID:true UsingOracleGTID:true UsingMariaDBGTID:false UsingPseudoGTID:false ReadBinlogCoordinates:vt-0000000300-bin.000001:76812 ExecBinlogCoordinates:vt-0000000300-bin.000001:76812 IsDetached:false RelaylogCoordinates:vt-0000000302-relay-bin.000002:77041 LastSQLError: LastIOError:"error reconnecting to master 'vt_repl@localhost:17300' - retry-time: 10  retries: 37" SecondsBehindPrimary:{Int64:0 Valid:false} SQLDelay:0 ExecutedGtidSet:64d59093-c795-11ed-93d1-000c296af92f:1-115 GtidPurged: GtidErrant: primaryExecutedGtidSet:64d59093-c795-11ed-93d1-000c296af92f:1-115 ReplicationLagSeconds:{Int64:0 Valid:false} DataCenter:zone1 Region: PhysicalEnvironment: ReplicationDepth:1 IsCoPrimary:false HasReplicationCredentials:true SemiSyncEnforced:false SemiSyncPrimaryEnabled:false SemiSyncReplicaEnabled:false SemiSyncPrimaryTimeout:1000000000000000000 SemiSyncPrimaryWaitForReplicaCount:1 SemiSyncPrimaryStatus:false SemiSyncPrimaryClients:0 SemiSyncReplicaStatus:false LastSeenTimestamp: IsLastCheckValid:true IsUpToDate:true IsRecentlyChecked:true SecondsSinceLastSeen:{Int64:0 Valid:false} IsCandidate:false PromotionRule:must_not IsDowntimed:false DowntimeReason: DowntimeOwner: DowntimeEndTimestamp: ElapsedDowntime:0s UnresolvedHostname: AllowTLS:false Problems:[] LastDiscoveryLatency:21.195247ms seed:false ReplicationGroupName: ReplicationGroupIsSinglePrimary:false ReplicationGroupMemberState: ReplicationGroupMemberRole: ReplicationGroupMembers:map[] ReplicationGroupPrimaryInstanceKey::0}, err - <nil>I0321 11:34:45.774751 1629626 orchestrator.go:238] Force discovered - &{Key:localhost:17301 InstanceAlias:zone1-0000000301 ServerID:1776132157 ServerUUID:6750ddf8-c795-11ed-9cce-000c296af92f Version:5.7
.41 VersionComment:MySQL Community Server (GPL) FlavorName: ReadOnly:true BinlogFormat:ROW BinlogRowImage:FULL LogBinEnabled:true LogReplicationUpdatesEnabled:true SelfBinlogCoordinates:vt-0000000301-bin.000001:75543 SourceKey:localhost:17300 SourceUUID:64d59093-c795-11ed-93d1-000c296af92f AncestryUUID:64d59093-c795-11ed-93d1-000c296af92f,6750ddf8-c795-11ed-9cce-000c296af92f IsDetachedPrimary:false ReplicationSQLThreadRuning:true ReplicationIOThreadRuning:false ReplicationSQLThreadState:1 ReplicationIOThreadState:2 HasReplicationFilters:false GTIDMode:ON SupportsOracleGTID:true UsingOracleGTID:true UsingMariaDBGTID:false UsingPseudoGTID:false ReadBinlogCoordinates:vt-0000000300-bin.000001:76812 ExecBinlogCoordinates:vt-0000000300-bin.000001:76812 IsDetached:false RelaylogCoordinates:vt-0000000301-relay-bin.000002:77041 LastSQLError: LastIOError:"error reconnecting to master 'vt_repl@localhost:17300' - retry-time: 10  retries: 37" SecondsBehindPrimary:{Int64:0 Valid:false} SQLDelay:0 ExecutedGtidSet:64d59093-c795-11ed-93d1-000c296af92f:1-115 GtidPurged: GtidErrant: primaryExecutedGtidSet:64d59093-c795-11ed-93d1-000c296af92f:1-115 ReplicationLagSeconds:{Int64:0 Valid:false} DataCenter:zone1 Region: PhysicalEnvironment: ReplicationDepth:1 IsCoPrimary:false HasReplicationCredentials:true SemiSyncEnforced:false SemiSyncPrimaryEnabled:false SemiSyncReplicaEnabled:true SemiSyncPrimaryTimeout:1000000000000000000 SemiSyncPrimaryWaitForReplicaCount:1 SemiSyncPrimaryStatus:false SemiSyncPrimaryClients:0 SemiSyncReplicaStatus:true LastSeenTimestamp: IsLastCheckValid:true IsUpToDate:true IsRecentlyChecked:true SecondsSinceLastSeen:{Int64:0 Valid:false} IsCandidate:false PromotionRule:neutral IsDowntimed:false DowntimeReason: DowntimeOwner: DowntimeEndTimestamp: ElapsedDowntime:0s UnresolvedHostname: AllowTLS:false Problems:[] LastDiscoveryLatency:28.368681ms seed:false ReplicationGroupName: ReplicationGroupIsSinglePrimary:false ReplicationGroupMemberState: ReplicationGroupMemberRole: ReplicationGroupMembers:map[] ReplicationGroupPrimaryInstanceKey::0}, err - <nil>I0321 11:34:45.775418 1629626 topology_recovery.go:209] topology_recovery: Waiting for 0 postponed functions
I0321 11:34:45.775780 1629626 postponed_functions.go:52] PostponedFunctionsContainer: waiting on 0 postponed functions
I0321 11:34:45.775803 1629626 postponed_functions.go:54] PostponedFunctionsContainer: done waiting
I0321 11:34:45.775809 1629626 topology_recovery.go:209] topology_recovery: Executed 0 postponed functions
I0321 11:34:45.775977 1629626 locks.go:470] Unlocking shard customer/-80 for action Orc Recovery with error Code: FAILED_PRECONDITION
no valid candidates for emergency reparent
E0321 11:34:45.777207 1629626 topology_recovery.go:803] Code: FAILED_PRECONDITION
no valid candidates for emergency reparent

```

**final status:**
<html>
<body>


Keyspace | Shard | Alias | Type | Tablet State | Hostname | Actions
-- | -- | -- | -- | -- | -- | --
commercelocal | 0 | zone1-100 | PRIMARY | SERVING | localhost |  
commercelocal | 0 | zone1-102 | RDONLY | SERVING | localhost |  
commercelocal | 0 | zone1-101 | REPLICA | SERVING | localhost |  
customerlocal | -80 | zone1-300 | PRIMARY | NOT_SERVING | localhost |  
customerlocal | -80 | zone1-302 | RDONLY | SERVING | localhost |  
customerlocal | -80 | zone1-301 | REPLICA | SERVING | localhost


</body>
</html>

**Comment by shizhibo2016** [comment]
Where is the status info?  vtadmin?

**Comment by isredstar** [comment]
vtadmin

**Comment by isredstar** [comment]
@shizhibo2016 When  the nodes are two replica, failover normally, but when the nodes are a rdonly and a replica, failover not normally

**Comment by shizhibo2016** [comment]
@isredstar  Which OS you use to run vitess?  Ubuntu,CentOS？

**Comment by isredstar** [comment]
CentOS 8 stream

**Comment by isredstar** [comment]
@shizhibo2016 one primary, one replica,  one rdonly can automated failover on your deployment.

**Comment by shizhibo2016** [comment]
Different DurabilityPolicy Different failover policy，But Why？

**Comment by isredstar** [comment]
**_semi_sync -_** This durability policy sets the number of required semi-sync ACKers to 1. It only allows Primary and Replica type servers to acknowledge semi sync. It returns NeutralPromoteRule for replica tablet types, MustNotPromoteRule for everything else.

**_none (default) -_** This durability policy does not set any semi-sync configurations. It returns NeutralPromoteRule for Primary and Replica tablet types, MustNotPromoteRule for everything else

**_cross_cell -_** This durability policy sets the number of required semi-sync ACKers to 1. It only allows Primary and Replica type servers from a different cell than the current primary to acknowledge semi sync. It returns NeutralPromoteRule for replica tablet types, MustNotPromoteRule for everything else.

**Comment by shizhibo2016** [comment]
So semi_sync policy must have 2 replica?
and cross_cell policy must have 1 replica from same cell and 1 from diffenrent cell ?

**Comment by isredstar** [comment]
That's true

**Comment by shizhibo2016** [comment]
@isredstar  Which version is your Vitess？
